safety-oracle.tex

```diff
@@ -0,0 +1,1552 @@
+\end{align}
+\end{proof}
+
+
+
+%---------------------------------------------------------------------------------------------------------------------
+%---------------------------------------------------------------------------------------------------------------------
+%---------------------------------------------------------------------------------------------------------------------
+%---------------------------------------------------------------------------------------------------------------------
+%---------------------------------------------------------------------------------------------------------------------
+%---------------------------------------------------------------------------------------------------------------------
+
+\subsubsection{Honest messages from majority clique members do not break the clique}
+
+
+\begin{lemma}
+$$
+\overline{v} \notin E(\sigma') \implies !L^H_M(Justification(\overline{m}))(\overline{v}) = !L^H_M(\sigma)(\overline{v})
+$$
+\end{lemma}
+
+\begin{proof}
+First, we show that $\overline{v}$ is not equivocating in $Justification(\overline{m})$.
+\begin{align}
+          &\overline{v} \notin E(\sigma') \\
+  \implies&\overline{v} \notin E(\sigma') \land Justification(\overline{m}) \subseteq \sigma \land \sigma \subset \sigma' \\
+  \implies&\overline{v} \notin E(\sigma') \land Justification(\overline{m}) \subset \sigma' \\
+  \implies&\overline{v} \notin E(\sigma') \land E(Justification(\overline{m})) \subseteq E(\sigma') \\
+  \implies&\overline{v} \notin E(Justification(\overline{m})) \\
+\end{align}
+
+Now, assume towards a contradiction:
+$$
+\exists m \in Justification(\overline{m}): Sender(m) = \overline{v} \land Later(m, !L^H_M(\sigma)(\overline{v}))
+$$
+\begin{align}
+  &\exists m \in Justification(\overline{m}): Sender(m) = \overline{v} \land Later(m, !L^H_M(\sigma)(\overline{v})) \\
+  \implies& \exists m \in Justification(\overline{m}): Sender(m) = \overline{v} \land Later(m, !L^H_M(\sigma)(\overline{v})) \land Justification(\overline{m}) \subseteq \sigma \\
+  \implies& \exists m \in \sigma: Sender(m) = \overline{v} \land Later(m, !L^H_M(\sigma)(\overline{v})) \\
+  \implies& \{m \in \sigma : Sender(m) = \overline{v} \land Later(m, !L^H_M(\sigma)(\overline{v}))\} \neq \emptyset \\
+  \implies& Later\_From(!L^H_M(\sigma)(\overline{v})), \overline{v}, \sigma) \neq \emptyset \\
+\end{align}
+Thus, this is clearly a contradiction, as the latest message in a state cannot have any later messages. Thus, we know that:
+$$
+\neg (\exists m \in Justification(\overline{m}): Sender(m) = \overline{v} \land Later(m, !L^H_M(\sigma)(\overline{v})))
+$$
+which is enough to complete the proof.
+\begin{align}
+          &\neg (\exists m \in Justification(\overline{m}): Sender(m) = \overline{v} \land Later(m, !L^H_M(\sigma)(\overline{v}))) \\
+  \implies&\nexists m \in Justification(\overline{m}): Sender(m) = \overline{v} \land Later(m, !L^H_M(\sigma)(\overline{v})) \\
+  \implies&\{m \in Just
```

Comment:
The function signature is wrong.

Reply:
and the order of params `σ` and `p` of `Clique` is reversed from the definition 7.16?